modules/tinymce/src/core/main/ts/keyboard/Nbsps.ts

```diff
@@ -149,22 +162,37 @@ const normalizeNbspInMiddleOfTextNode = (node: Text): boolean => {
   }
 };
 
-const normalizeNbspAtEnd = (root: SugarElement<Node>, node: Text): boolean => {
+const normalizeNbspAtEnd = (root: SugarElement<Node>, node: Text, makeNbsp: boolean): boolean => {
   const text = node.data;
   const lastPos = CaretPosition(node, text.length - 1);
-  if (isNbspAt(text, text.length - 1) && !needsToBeNbsp(root, lastPos)) {
+  if (isNbspAt(text, text.length - 1) && !needsToBeNbsp(root, lastPos) && !makeNbsp) {
     node.data = text.slice(0, -1) + ' ';
     return true;
+  } else if (makeNbsp && isWhiteSpaceAt(text, text.length - 1) && needsToBeNbspRight(root, lastPos)) {
+    node.data = text.slice(0, -1) + Unicode.nbsp;
+    return true;
   } else {
     return false;
   }
 };
 
-const normalizeNbsps = (root: SugarElement<Node>, pos: CaretPosition): Optional<CaretPosition> => Optional.some(pos).filter(hasNbsp).bind((pos) => {
-  const container = pos.container() as Text;
-  const normalized = normalizeNbspAtStart(root, container) || normalizeNbspInMiddleOfTextNode(container) || normalizeNbspAtEnd(root, container);
-  return normalized ? Optional.some(pos) : Optional.none();
-});
+const normalizeNbsps = (root: SugarElement<Node>, pos: CaretPosition): Optional<CaretPosition> =>
+  Optional.some(pos).filter(hasNbsp).fold(
+    () => {
+      if (needsToBeNbsp(root, pos) && NodeType.isText(pos.container())) {
+        const container = pos.container() as Text;
+        const normalized = normalizeNbspAtStart(root, container, true) || normalizeNbspAtEnd(root, container, true);
+        return normalized ? Optional.some(pos) : Optional.none();
```

Comment:
Up to you if you do this, but this could be using the `Optionals.someIf(normalized, pos)` helper here and below to reduce the bundle size (albeit very marginally).